### PR TITLE
FWT-8 Looks Very Solid Still

### PR DESCRIPTION
**Pull request by DiegoLHendrix**

### CODE REVIEW

File: include/dev/ACS71240.hpp [hunk 1]
```diff
@@ -13,29 +13,24 @@ namespace LVSS {
...
 
     /**
-      * Get the current detected by the INA
+      * Get the current detected by the ACS71240
       *
       * @return The current
       */
-    uint32_t readCurrent();
+    int32_t readCurrent();
+    uint32_t readCounts();
```
Comment (mjmagee991) [inline include/dev/ACS71240.hpp]: Delete this function. You were just using it for debugging

File: src/dev/ACS71240.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace LVSS {
+
+ACS71240::ACS71240(IO::ADC& adc0) : ADC(adc0) {}
+
+int32_t ACS71240::readCurrent() {
+    //Gets adcCounts from adc
+    int32_t adcCounts = ADC.readRaw();
+
+    int32_t current = (((adcCounts - 1970) * 3300) / 180);
```
Comment (mjmagee991) [inline src/dev/ACS71240.cpp:12]: Add a comment explaining this equation

File: targets/REV3-ACS71240/main.cpp [hunk 3]
```diff
@@ -25,12 +25,14 @@ int main() {
     log::LOGGER.setLogLevel(log::Logger::LogLevel::INFO);
 
     //ADC
-    IO::ADC& adc0 = IO::getADC<IO::Pin::PA_1>();
+    IO::ADC& adc0 = IO::getADC<IO::Pin::A1>();
```
Comment (mjmagee991) [inline targets/REV3-ACS71240/main.cpp]:
```suggestion
    IO::ADC& adc0 = IO::getADC<IO::Pin::PA1>();
```

File: targets/REV3-ACS71240/main.cpp [hunk 4]
```diff
@@ -25,12 +25,14 @@ int main() {
...
 
-    LVSS::INA138 ina138(adc0);
+    //Create ACS71240 instance
+    LVSS::ACS71240 acs71240(adc0);
 
     while (1) {
-        uart.printf("ADC0: %d\r\n mA", ina138.readCurrent());
-        time::wait(500);
+        uart.printf("ADC0: %dmA\r\n", acs71240.readCurrent());
+        uart.printf("Counts: %d\r\n", acs71240.readCounts());
```
Comment (mjmagee991) [inline targets/REV3-ACS71240/main.cpp]:
```suggestion
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A couple things to clean up

File: include/dev/ACS71240.hpp [hunk 5]
```diff
@@ -13,29 +13,23 @@ namespace LVSS {
...
      * Constructor for current sensing class
      */
-    INA138(IO::ADC& adc0);
+    ACS71240(IO::ADC& adc0);
 
     /**
-      * Get the current detected by the INA
+      * Get the current detected by the ACS71240
       *
       * @return The current
```
Comment (mjmagee991) [inline include/dev/ACS71240.hpp]:
```suggestion
      * @return The current in mA
```

File: src/dev/ACS71240.cpp (hunk 2 above)
Comment (mjmagee991) [reply src/dev/ACS71240.cpp:12]: The comment should explain where the 3300 and 180 came from

File: targets/REV3-ACS71240/main.cpp [hunk 6]
```diff
@@ -27,10 +27,10 @@ int main() {
...
     IO::ADC& adc0 = IO::getADC<IO::Pin::PA_1>();
 
-    LVSS::INA138 ina138(adc0);
+    //Create ACS71240 instance
+    LVSS::ACS71240 acs71240(adc0);
 
     while (1) {
-        uart.printf("ADC0: %d\r\n mA", ina138.readCurrent());
-        time::wait(500);
+        uart.printf("ADC0: %dmA\r\n", acs71240.readCurrent());
```
Comment (mjmagee991) [inline targets/REV3-ACS71240/main.cpp:34]: Add the wait back in because this will be way too fast to read

**Review by mjmagee991 — CHANGES_REQUESTED** [review]

**Review by mjmagee991 — APPROVED** [review]
Great work!

**Review by tmb5932 — APPROVED** [review]
# This _gas_ frfr

**Pull request merged**